validator/lib/validateTileset.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var Promise = require('bluebird');
+
+module.exports = validateTileset;
+
+/**
+ * Walks down the tree represented by the JSON object and checks if it is a valid tileset.
+ *
+ * @param {Object} tileset The JSON object representing the tileset.
+ * @return {Promise} A promise that resolves with two parameters - (1) a boolean for whether the tileset is valid
+ *                                                                 (2) the error message if the tileset is not valid.
+ *
+ */
+function validateTileset(tileset) {
+    return new Promise(function(resolve) {
+        validateNode(tileset.root, tileset.geometricError, resolve);
+    });
+}
+
+function validateNode(root, parentGeometricError, resolve) {
+    var stack = [];
+    stack.push({'node': root, 'parentError': parentGeometricError});
+
+    while (stack.length > 0) {
+
+        var obj = stack.pop();
```

Comment:
Create variables for `node` and `parent` here instead of accessing `obj['node']` and `obj['parentError']` all the time below. Also use dot notation for accessing members:

```javascript
var node = obj.node;
var parent = obj.parent;
```


For naming, it might be better if `obj` -> `node` and `node` -> `tile`.